src/Raven.Server/Documents/Handlers/Admin/AdminConfigurationHandler.cs

```diff
@@ -112,6 +127,21 @@ public async Task PutSettings()
             NoContentStatus(HttpStatusCode.Created);
         }
 
+        private static List<string> GetUpdatedSettingsKeys(Dictionary<string, string> currentSettings, Dictionary<string, string> settingsToUpdate)
+        {
+            var updatedSettings = new List<string>();
+
+            foreach (var settingToUpdate in settingsToUpdate)
+            {
+                currentSettings.TryGetValue(settingToUpdate.Key, out var currentSettingValue);
```

Comment:
if it does not exist then it should be added immediately to the list, comparison should be done only if they key exists

imagine  situation when you do not have a key in settings (and then the default is 100 for that key) and you are putting `null` (assuming it is acceptable). Then instead of adding it to the list, you are comparing `null` with `null`

Reply:
Changed